packages/astro/src/core/create-vite.ts

```diff
@@ -55,6 +55,7 @@ export async function createVite(
 		logLevel: 'warn', // log warnings and errors only
 		optimizeDeps: {
 			entries: ['src/**/*'], // Try and scan a user’s project (won’t catch everything),
+			exclude: ['node-fetch'],
```

Comment:
Here's what I think you should do differently.....















j/k, lgtm